waspc/packages/deploy/src/providers/fly/deploy/deploy.ts

```diff
@@ -0,0 +1,119 @@
+import { exit } from 'process';
+import { $, cd, fs } from 'zx';
+import {
+	cdToClientBuildDir,
+	cdToServerBuildDir,
+	displayWaspRocketImage,
+	makeIdempotent,
+	getCommandHelp,
+	waspSays,
+} from '../helpers/helpers.js';
+import {
+	clientTomlExistsInProject,
+	copyLocalClientTomlToProject,
+	copyLocalServerTomlToProject,
+	copyProjectClientTomlLocally,
+	copyProjectServerTomlLocally,
+	getInferredBasenameFromClientToml,
+	getInferredBasenameFromServerToml,
+	getTomlFilePaths,
+	serverTomlExistsInProject,
+} from '../helpers/tomlFileHelpers.js';
+import { DeployOptions } from './DeployOptions.js';
+import { createDeploymentInfo, DeploymentInfo } from '../DeploymentInfo.js';
+import { flySetupCommand } from '../index.js';
+import { secretExists } from '../helpers/flyctlHelpers.js';
+
+export async function deploy(options: DeployOptions): Promise<void> {
+	waspSays('Deploying your Wasp app to Fly.io!');
+
+	const buildWasp = makeIdempotent(async () => {
+		if (options.skipBuild) {
+			return;
+		}
+
+		waspSays('Building your Wasp app...');
+		cd(options.waspProjectDir);
+		await $`${options.waspExe} build`;
```

Comment:
Nit: we could have abstracted all `wasp` exe invocations with some function `runWaspCli('build')`, but on the other hand this is super explicit which is nice as well.

Probably over time when we add the next provider we'll figure out if we need it.

Reply:
True. It would still need access to the CLI options object somehow though to get the path, but yeah we could have wrapped it some how. I expect we will find lots of opportunities to refactor once we add the second provider, so I will keep that in mind for next time! 👍🏻